From e80fa15c0bfe018687ea7b95c54db907e094830e Mon Sep 17 00:00:00 2001
From: Viswanath Kraleti <viswanath.kraleti@oss.qualcomm.com>
Date: Fri, 10 Jul 2026 18:51:29 +0530
Subject: [PATCH] Add Nord support

Signed-off-by: Viswanath Kraleti <viswanath.kraleti@oss.qualcomm.com>
Upstream-Status: Pending
---
 qcom-metadata.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/qcom-metadata.dts b/qcom-metadata.dts
index 333464a..87a9727 100644
--- a/qcom-metadata.dts
+++ b/qcom-metadata.dts
@@ -28,10 +28,18 @@
 			msm-id = <0x000002b5>;
 		};
 
+		nord {
+			msm-id = <0x00000288>;
+		};
+
 		purwa {
 			msm-id = <0x000002c7>;
 		};
 
+		qam8x97p {
+			msm-id = <0x000002b2>;
+		};
+
 		qcm6490 {
 			msm-id = <0x000001f1>;
 		};
