From cd7e8affaef62340405a6e477ffa4c4b96e3e6c5 Mon Sep 17 00:00:00 2001
From: Viswanath Kraleti <viswanath.kraleti@oss.qualcomm.com>
Date: Fri, 10 Jul 2026 18:51:29 +0530
Subject: [PATCH] Add Nord support

Signed-off-by: Viswanath Kraleti <viswanath.kraleti@oss.qualcomm.com>
Upstream-Status: Pending
---
 qcom-metadata.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/qcom-metadata.dts b/qcom-metadata.dts
index 17b5028..878663a 100644
--- a/qcom-metadata.dts
+++ b/qcom-metadata.dts
@@ -24,10 +24,18 @@
 			msm-id = <0x00000294>;
 		};
 
+		nord {
+			msm-id = <0x00000288>;
+		};
+
 		purwa {
 			msm-id = <0x000002c7>;
 		};
 
+		qam8x97p {
+			msm-id = <0x000002b2>;
+		};
+
 		qcm6490 {
 			msm-id = <0x000001f1>;
 		};
-- 
2.34.1

